src/config.rs

```diff
@@ -0,0 +1,136 @@
+//! Provides a simple way of reading configuration files.
+//!
+//! Reads a TOML config file for the IPC Agent and deserializes it in a type-safe way into a
+//! [`Config`] struct. Usage:
+//!
+//! ```
+//! let c: Config = Config::from_file("config/example.toml");
+//! ```
+
+use std::collections::HashMap;
+use std::fmt::Formatter;
+use std::fs;
+use std::str::FromStr;
+
+use anyhow::Result;
+use fvm_shared::address::Address;
+use ipc_sdk::subnet_id::SubnetID;
+use serde::de::{Error, SeqAccess};
+use serde::{Deserialize, Deserializer};
+use url::Url;
+
+/// The top-level struct representing the config. Calls to [`Config::from_file`] deserialize into this
+/// struct.
+#[derive(Deserialize)]
+pub(crate) struct Config {
+    pub subnets: HashMap<String, Subnet>,
+}
+
+/// Represents a subnet declaration in the config.
+#[derive(Deserialize)]
+pub struct Subnet {
+    #[serde(deserialize_with = "deserialize_path")]
+    path: SubnetID,
```

Comment:
Why not just call it `subnet_id`? Could multiple subnets have the same`subnet_id`? If possible, I assume their `path` would be different, but they can have the same `subnet_id`? Actually what does the `path` mean, maybe attach the design doc could help.

Reply:
Yes, I think that would be preferable. Perhaps, using simply `id` because the field is already under a `subnet` section in the config. @adlrocha What do you think of the name change?